### PR TITLE
feat(metrics):  increase maximum dimensions to 29

### PR DESCRIPTION
**Pull request by sthuber90**
## Description of your changes

Amazon added support to up 30 dimensions per metric. In this PR we increase this limit

### How to verify this change

Review the code changes and the results of the automated builds.

### Related issues, RFCs

**Issue number:**  #1044 

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [ ] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [ ] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/metrics/tests/unit/Metrics.test.ts
Comment (dreamorosi) [inline packages/metrics/tests/unit/Metrics.test.ts:462]: Could you please move this assertion at the end of the test? This file specifically is a bit of an outlier, but generally speaking we try to do 1/ Setup, 2/ Action, 3/ Assess

Comment (dreamorosi) [reply packages/metrics/tests/unit/Metrics.test.ts:462]: I just saw that the assertion is used at the beginning in other test cases of this file, we will be changing this in the context of #163 to make them more uniform to the other two utils.

You can choose to ignore the comment or address it, it won't be a blocker for this PR.

**Review by dreamorosi — APPROVED** [review]
Thank you for opening the PR.

I'm going to ask for a review to another maintainer as we have a 2 approvals model for each PR.

Note for the next maintainer: I have run e2e tests and the result was successful:
image

File: packages/metrics/tests/unit/Metrics.test.ts
Comment (sthuber90) [reply packages/metrics/tests/unit/Metrics.test.ts:462]: As you already said, I added it at the beginning just as it's the case in other tests. If I hadn't seen this, I would have used the `fail('reason')` before the `catch` block but that's something to be adressed in the issue that you've referenced

**Review by ijemmy — APPROVED** [review]
LGTM, thanks for your contribution @sthuber90   :)

**Pull request merged**
